Travis build deploy

dehall opened the pull request:
Adds a deploy step to the travis run script, based on instructions here: https://docs.travis-ci.com/user/deployment/pages/

Also adds the gh-pages URL to the package.json so that URLs in the build work as expected.

I tested this on a fork under my own account and it worked, see: https://github.com/dehall/pathways/tree/gh-pages and https://dehall.github.io/pathways/

The change to package.json and yarn.lock is to use the version of fhir-visualizers which is now on NPM, to fix this error: https://travis-ci.org/mcode/pathways/builds/646984479

dehall committed:
Add build & deploy to travis

dehall committed:
add gh pages url to package.json

dehall committed:
push fhir-viz to npm to reduce headache

radamson reviewed (approved):
LGTM.

We should stop tracking the `build` directory in fhir-visualizers now that it's on npm.

dehall committed:
replace BrowserRouter with HashRouter

dehall committed:
add static list of pathways

dehall committed:
replace pathwaysService param in travis script

dehall committed:
update deploy script with unescaped url

dehall commented:
Based on offline conversations with @blangley28 , I updated this so that it would work (as much as possible) via the GH pages. Unfortunately there's still a blocking error, and that's not something I can resolve in code or deployment scripts here.

Latest changes include:
 - changing the react-router to use a HashRouter instead of a BrowserRouter, so that the demo URL works on gh-pages. Because gh-pages uses a pretty simple URL matching scheme, trying to hit `/demo` would try to load an actual file /demo, and it wouldn't work. Now with a hash router, the demo URL is `/#/demo`. A little ugly but works.
 - Adding a static list of pathways that we can fetch using the existing pathway service. It's .html because of discrepancies between gh-pages and react-scripts. Added a note about that in a separate file
 - as part of deploy, rewrite the config file so that it points to the static list of pathways


So, the idea with all that is you should be able to hit https://dehall.github.io/pathways/#/demo or SMART launch to https://dehall.github.io/pathways/ and it will work. And it's so close. The last remaining piece is the CQL-to-ELM translation service. For one thing it's on our internal network, but it's also HTTP only, not HTTPS, so the browser is going to block requests on an HTTPS page from going to an HTTP resource. No way around that that I can see. I'm going to explore other possibilities around that, but for now there's nothing I can do about that and I don't think that should hold up this PR

dehall committed:
use public cql-to-elm service
